test/integration/kwild_test.go

```diff
@@ -307,52 +309,63 @@ func TestKwildEthDepositOracleIntegration(t *testing.T) {
 		t.Parallel()
 	}
 
-	opts := []integration.HelperOpt{
-		integration.WithBlockInterval(time.Second),
-		integration.WithValidators(4),
-		integration.WithNonValidators(0),
-		integration.WithGanache(),
-		integration.WithGas(),
-		integration.WithEthDepositOracle(true),
+	type testcase struct {
+		name          string
+		numValidators int
+		serviceNames  []string
+	}
+
+	testcases := []testcase{
+		{"single-node", 1, singleNodeServices},
+		{"multi-node", 4, allServices},
 	}
 
 	testDrivers := strings.Split(*drivers, ",")
 	for _, driverType := range testDrivers {
-		t.Run(driverType+"_driver", func(t *testing.T) {
-			ctx := context.Background()
-
-			helper := integration.NewIntHelper(t, opts...)
-			helper.Setup(ctx, allServices)
-
-			// get deployer
-			ctxMiner, cancel := context.WithCancel(ctx)
-			defer cancel()
-			deployer := helper.EthDeployer(false)
-			err := deployer.KeepMining(ctxMiner)
-			require.NoError(t, err)
-
-			// Get the user driver
-			userDriver := helper.GetUserDriver(ctx, "node0", driverType, deployer)
-
-			// Node0 Driver
-			// node0Driver := helper.GetOperatorDriver(ctx, "node0", driverType)
-
-			// Approve the deposit
-			// specifications.ApproveSpecification(ctx, t, userDriver)
-
-			// Deposit the amount to the escrow
-			amount := big.NewInt(10)
-			specifications.DepositSuccessSpecification(ctx, t, userDriver, amount)
-
-			// Deposit Failure
-			// specifications.DepositFailSpecification(ctx, t, userDriver)
-
-			// Deploy DB without enough funds
-			specifications.DeployDbInsufficientFundsSpecification(ctx, t, userDriver)
-
-			// Deploy DB with enough funds
-			specifications.DeployDbSuccessSpecification(ctx, t, userDriver)
-		})
+		for _, tc := range testcases {
+			t.Run(tc.name+"_"+driverType+"_driver", func(t *testing.T) {
+				ctx := context.Background()
+				opts := []integration.HelperOpt{
+					integration.WithBlockInterval(time.Second),
+					integration.WithValidators(tc.numValidators),
+					integration.WithNonValidators(0),
+					integration.WithGanache(),
+					integration.WithGas(),
+					integration.WithEthDepositOracle(true),
+				}
+				helper := integration.NewIntHelper(t, opts...)
+				helper.Setup(ctx, tc.serviceNames)
+
+				// get deployer
+				ctxMiner, cancel := context.WithCancel(ctx)
+				defer cancel()
+				deployer := helper.EthDeployer(false)
+				err := deployer.KeepMining(ctxMiner)
+				require.NoError(t, err)
+
+				// Get the user driver
+				userDriver := helper.GetUserDriver(ctx, "node0", driverType, deployer)
+
+				// Node0 Driver
+				// node0Driver := helper.GetOperatorDriver(ctx, "node0", driverType)
+
+				// Approve the deposit
+				// specifications.ApproveSpecification(ctx, t, userDriver)
```

Comment:
Not sure why I didn't just remove this when I commented it out.  Probably OK to remove if we agree there's no business doing ERC20 stuff in test specifications.  Every time I see "approve" I wonder if it's a validator/vote thing or erc20 allowance business...